provider/pkg/provider/provider.go

```diff
@@ -1806,6 +1807,13 @@ func (k *kubeProvider) Create(
 	initialized, awaitErr := await.Creation(config)
 	if awaitErr != nil {
 		if req.GetPreview() {
+			if apierrors.IsForbidden(awaitErr) {
```

Comment:
I have opted to still return an error if preview fails in SSA mode. Should we instead ignore the error, and essentially do a client-side diff against the stored state?

Reply:
Yeah, I think it would be better to show a preview of the inputs rather than erroring. For SSA mode, it probably also makes sense to log an info message to indicate that it's an input-only preview rather than a SSA dry-run create.

Reply:
Updated to not return an error for `pulumi preview`. Have also added tests for this.